packages/canon/src/components/DataTable/Pagination/DataTablePagination.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { forwardRef } from 'react';
+import { Text } from '../../Text';
+import { DataTablePaginationProps } from './types';
+import { IconButton } from '../../IconButton';
+import clsx from 'clsx';
+import { Select } from '../../Select';
+import { useDataTable } from '../Root/DataTableRoot';
+
+/** @public */
+const DataTablePagination = forwardRef(
+  (
+    props: DataTablePaginationProps,
+    ref: React.ForwardedRef<HTMLDivElement>,
+  ) => {
+    const { className, ...rest } = props;
+    const { table } = useDataTable();
+    const pageIndex = table?.getState().pagination.pageIndex;
+    const pageSize = table?.getState().pagination.pageSize;
+
+    return (
+      <div
+        ref={ref}
+        className={clsx('canon-DataTablePagination', className)}
+        {...rest}
+      >
+        <div className="canon-DataTablePagination--left">
+          {!table.options.manualPagination && (
+            <Select
+              name="pageSize"
+              size="small"
+              placeholder="Show 10 results"
+              options={[
```

Comment:
I assume we want there configurable later on

Reply:
Yes this component will evolve quite a bit. This first version is meant as a starter for the catalog page. We'll need to gather more use cases to understand what the props will look like.